pkg/apis/v1alpha5/machine_disruption_gate.go

```diff
@@ -0,0 +1,94 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha5
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/robfig/cron/v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/clock"
+	"knative.dev/pkg/logging"
+)
+
+var AllowedActions sets.String = sets.NewString("Consolidation", "Emptiness", "Expiration", "Drift")
+var CrontabTemplateFormat = "TZ=UTC %s"
+
+// MachineDisruptionGateSpec is the top level field for MachineDisruptionGates.
+type MachineDisruptionGateSpec struct {
+	// Schedules represent times where disruption is prohibited by Karpenter.
+	// Schedules will be parsed in UTC. If the current time is captured
+	// by any of the schedules, the Machine Disruption Gate is considered active.
+	Schedules []Schedule `json:"schedules"`
```

Comment:
Should this be plural or not? I'm not sure.

Reply:
If I were to consider this `Schedule` I would make the type something other than `[]Schedule`, maybe `[]Period` or `[]Interval`, which in each case I'd like to name the field itself the plural form of the word. Thoughts?